valkyrie/lib/valkyrie/persistence/solr/mapper.rb

```diff
@@ -29,23 +36,49 @@ def to_h
 
     private
 
+      def attribute_hash
+        properties.each_with_object({}) do |property, hsh|
+          SolrMapperValue.for(Property.new(property, object.attributes[property])).result.apply_to(hsh)
+        end
+      end
+
+      def properties
+        object.attributes.keys - [:id, :created_at, :updated_at]
+      end
+
+      ##
+      # A container object for holding a `key`, `value, and `scope` of a value
+      #   in a model together for casting.
       class Property
         attr_reader :key, :value, :scope
+        # @param key [Symbol] Property identifier.
+        # @param value [Object] Value or list of values which are underneath the
+        #   key.
+        # @param scope [Object] The object or point where the key and values
+        #   came from.
         def initialize(key, value, scope = [])
           @key = key
           @value = value
           @scope = scope
         end
       end
 
+      ##
+      # Represents a row in the solr document, used for isolating logic around
```

Comment:
Solr doesn't have rows, so this seems forced, but it might require a class name change.